akka-remote/src/main/scala/akka/remote/security/provider/AESCounterBuiltinRNG.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2016-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.security.provider
+
+import java.security.Key
+import java.util.Random
+import java.util.concurrent.locks.ReentrantLock
+import javax.crypto.Cipher
+
+/**
+ * INTERNAL API
+ * This class is a Scala implementation of AESCounterRNG algorithm
+ * patterned after org.uncommons.maths.random by Daniel Dyer (Apache License 2.0)
+ *
+ * Non-linear random number generator based on the AES block cipher in counter mode.
+ * Uses the seed as a key to encrypt a 128-bit counter using AES(Rijndael).
+ *
+ * Keys larger than 128-bit for the AES cipher require
+ * the inconvenience of installing the unlimited strength cryptography policy
+ * files for the Java platform.  Larger keys may be used (192 or 256 bits) but if the
+ * cryptography policy files are not installed, a
+ * java.security.GeneralSecurityException will be thrown.
+ *
+ * NOTE: THIS CLASS IS NOT SERIALIZABLE
+ */
+
+class AESCounterBuiltinRNG(val seed: Array[Byte]) extends Random {
+  private val COUNTER_SIZE_BYTES = 16
+  private val BITWISE_BYTE_TO_INT = 0x000000FF
+
+  // mutable state below, concurrent accesses need lock
+  private val lock = new ReentrantLock
+  private val counter: Array[Byte] = new Array[Byte](COUNTER_SIZE_BYTES)
+  private var index: Int = 0
+  private var currentBlock: Array[Byte] = null
+
+  private val cipher = Cipher.getInstance("AES/ECB/NoPadding")
```

Comment:
wait, doesn't Cipher has an AES/CTR/NoPadding anyway? Why hand-build the counter? CTR just XORs the plaintext with the current counter output. I.e. you should just continuously encrypt the all-zero block and take its output as the random value. Also, I would prefer if this is automatically reseeded from time to time from the SecureRandom, i.e. after a certain number of outputs.